weave-js/src/components/Panel2/PanelObject.tsx

```diff
@@ -222,10 +222,19 @@ export const PanelObject: React.FC<PanelObjectProps> = props => {
             childNode={childNode}
             childType={propertyTypes[k]!}
             updateInput={updateInput}
+            readOnly={updateInputFromProps == null}
```

Comment:
curious why you didn't just make `updateInput` null if updateInputFromProps is null?, then linking is conditioned based on if `updateInput` is null in props. The saves you the extra param since they are equal

Reply:
nice -- changed in next commit